lib/nerves_hub_link/health_check/default_report.ex

```diff
@@ -0,0 +1,126 @@
+defmodule NervesHubLink.HealthCheck.DefaultReport do
+  alias NervesHubLink.HealthCheck.Report
+  require Logger
+  @behaviour NervesHubLink.HealthCheck.Report
+
+  @impl Report
+  def timestamp do
+    DateTime.utc_now()
+    |> DateTime.to_iso8601()
+  end
+
+  @impl Report
+  def metadata do
+    # A lot of typical metadata is included in the join
+    # we can skip that here
+    # NervesHub is responsible for joining that into the stored data
+    metadata_from_config()
+  end
+
+  @impl Report
+  def alarms do
+    for {id, description} <- :alarm_handler.get_alarms(), into: %{} do
+      try do
+        {inspect(id), inspect(description)}
+      catch
+        _, _ ->
+          {"bad alarm term", ""}
+      end
+    end
+  end
+
+  @impl Report
+  def metrics do
+    [
+      metrics_from_config(),
+      cpu_temperature(),
+      load_averages(),
+      memory()
+    ]
+    |> Enum.reduce(%{}, &Map.merge/2)
+  end
+
+  @impl Report
+  def peripherals do
+    peripherals_from_config()
+  end
+
+  defp vof({mod, fun, args}), do: apply(mod, fun, args)
+  defp vof(val), do: val
+
+  defp metadata_from_config do
+    case Application.get_env(:nerves_hub_link, :health, [])[:metadata] do
+      nil ->
+        %{}
+
+      metadata ->
+        for {key, val_or_fun} <- metadata, into: %{} do
+          {inspect(key), vof(val_or_fun)}
+        end
+    end
+  end
+
+  defp metrics_from_config do
+    case Application.get_env(:nerves_hub_link, :health, [])[:metrics] do
+      nil ->
+        %{}
+
+      metrics ->
+        for {key, val_or_fun} <- metrics, into: %{} do
+          {inspect(key), vof(val_or_fun)}
+        end
+    end
+  end
+
+  defp peripherals_from_config do
+    case Application.get_env(:nerves_hub_link, :health, [])[:peripherals] do
+      nil ->
+        %{}
+
+      peripherals ->
+        for {key, val_or_fun} <- peripherals, into: %{} do
+          {inspect(key), vof(val_or_fun)}
+        end
+    end
+  end
+
+  defp cpu_temperature do
+    with {:ok, content} <- File.read("/sys/class/thermal/thermal_zone0/temp"),
+         {millidegree_c, _} <- Integer.parse(content) do
+      %{cpu_temp: millidegree_c / 1000}
+    else
+      _ -> cpu_temperature_rpi()
+    end
+  end
+
+  defp cpu_temperature_rpi do
+    with {result, 0} <- System.cmd("/usr/bin/vcgencmd", ["measure_temp"]) do
+      %{"temp" => temp} = Regex.named_captures(~r/temp=(?<temp>[\d.]+)/, result)
+      {temp, _} = Integer.parse(temp)
+      %{cpu_temp: temp}
+    else
+      _ -> %{}
+    end
+  end
+
+  defp load_averages do
+    with {:ok, data_str} <- File.read("/proc/loadavg"),
+         [min1, min5, min15, _, _] <- String.split(data_str, " ") do
+      %{load_1min: min1, load_5min: min5, load_15min: min15}
```

Comment:
Why remove the load parsing into numbers? Metrics are supposed to be numbers, type-wise?

Reply:
you were change the percentages into just 0 or 1, and then I thought about sending up floats, but after talking with Jon, I opted for the parsing of this on hub, as hub will need to do normalization.

Reply:
I'm not tied to strings, we can send floats. We will likely want to also do normalization in hub too.

Reply:
Ah, that's a goof on my part to use Integers. The typing is `number()` specifically to support float and int which also works with JSON.